vllm/scripts.py

```diff
@@ -25,6 +26,13 @@ def signal_handler(sig, frame):
 
 
 def serve(args: argparse.Namespace) -> None:
+    if args.config:
+        config_parser = configparser.ConfigParser()
+        config_parser.read(args.config)
+
+        for key, value in config_parser.items():
+            setattr(args, key, value)
+
```

Comment:
if we have `--config`, make sure we don't have any other commandline args?

Reply:
wait, after a second though, it is not clear to me if we should make it mutually exclusive

Reply:
for example. if I have a config that has lots of options, I might want to simply reuse it and override some values from the commandline, e.g. `vllm serve --config config.yaml -tp 2`

Reply:
@youkaichao  howa bout this?

lets not make it exclusive to allow flexibility, BUT also raise a hard interrupt in case an option is specified in both cli and config files. this will prevent users from unknowingly launching their server with stale configs/args...

Reply:
or print an INFO level logging, telling users explicitly that some arg value from commandline takes precedence ?

Reply:
deal 🤝

Reply:
done. made cli the override